containers/hawkular-metrics-index-truncator/src/main/java/org/hawkular/openshift/truncator/OpenshiftClientWrapper.java

```diff
@@ -0,0 +1,167 @@
+        client.getAuthorizationContext().setToken(token);
+        this.client = client;
+    }
+
+    public void scaleReplicaitonController(String name, int replicas) {
+        IReplicationController replicationController = new ResourceFactory(client).create("v1", ResourceKind.REPLICATION_CONTROLLER);
+        ((ReplicationController) replicationController).setName(name);
+        ((ReplicationController) replicationController).setNamespace(this.namespace);
+        replicationController.accept(new CapabilityVisitor<IScalable, IScale>() {
+            @Override
+            public IScale visit(IScalable capability) {
+                return capability.scaleTo(replicas);
+            }
+        }, null);
+    }
+
+    public int getReplicationControllerScale(String name) {
+        IReplicationController p = client.get(ResourceKind.REPLICATION_CONTROLLER, name, this.namespace);
+        return p.getReplicas();
+    }
+
+    public Pod[] getPodsWithPrefix(String prefix) {
+        ArrayList<Pod> pods = new ArrayList<>();
+        Pattern prefixPattern = Pattern.compile("^" + prefix + ".*");
+        List<Pod> list = client.list(ResourceKind.POD, this.namespace);
+        for (Pod pod : list) {
+            if (prefixPattern.matcher(pod.getName()).matches() && "Running".equals(pod.getStatus())) {
+                pods.add(pod);
+            }
+        }
+        return pods.toArray(new Pod[0]);
+    }
+
+    private int getMaxPartitionOfTable(String table, Pod cassandraPod) throws java.lang.InterruptedException {
+        // We are using exec nodetool because we don't have access to JMX metrics to determine
+        // the estimated size of partitions, we use estimated max of tablehistograms.
+        String[] echoCommand = {NODETOOL_EXECUTABLE, "tablehistograms", "hawkular_metrics", table};
+        PartitionSizeListener partitionSizeListener = new PartitionSizeListener();
+        cassandraPod.accept(new CapabilityVisitor<IPodExec, IStoppable>() {
+            @Override
+            public IStoppable visit(IPodExec capability) {
+                return capability.start(partitionSizeListener, null, echoCommand);
+            }
+        }, null);
+        partitionSizeListener.await(execTimeOut, TimeUnit.SECONDS);
+        return partitionSizeListener.getPartitionSize();
+    }
+
+    public boolean hasLargePartition(int threshold) throws java.lang.InterruptedException {
+        Pod[] pods = this.getPodsWithPrefix("hawkular-cassandra");
+        if (pods.length > 0) {
+            Pod cassandra = pods[0];
+            int maxMetricIndexSize = getMaxPartitionOfTable("metrics_idx", cassandra);
+            int maxTagsIndexSize = getMaxPartitionOfTable("metrics_tags_idx", cassandra);
+            int maxPartition = Math.max(maxMetricIndexSize, maxTagsIndexSize);
+            return maxPartition > threshold;
+        } else {
+            log.error("No cassandra pods available, Is Hawkular Metrics running?");
```

Comment:
Perhaps this error should mention Cassandra in particular, as it occurs when Cassandra is not running